modules/core/src/main/java/org/apache/ignite/internal/maintenance/MaintenanceProcessor.java

```diff
@@ -120,6 +122,36 @@ public MaintenanceProcessor(GridKernalContext ctx) {
         return oldTask;
     }
 
+    /** {@inheritDoc} */
+    @Nullable @Override public MaintenanceTask registerMaintenanceTask(
```

Comment:
What if method **MaintenanceProcessor#registerMaintenanceTask(MaintenanceTask)** and **MaintenanceProcessor#registerMaintenanceTask(MaintenanceTask, Function)** call a private **MaintenanceProcessor#registerMaintenanceTask0(MaintenanceTask, @nullable Function)**? Then we can get rid of code duplication.